elastic/kibana
# [Alerting] Track number of queries, query and search time during each rule execution.

Resolves https://github.com/elastic/kibana/issues/124809

## Summary

* Wraps the search function in the ES client provided to rule executors with logic to keep track of the number of queries performed per execution and the total search and query time, where total query time is retrieved from the ES response body (`took` field) and total search time includes any network latency and time spent serializing/deserializing the request/response
* Adds debug logging to log the rule id, name, type and query body
* Adds this information to the `execute` event log document under `rule.execution.metrics`

Note that this does not touch the abortable search function as the plan is to roll in the abort signal with this wrapped search function and remove the parallel search function. Related issue: https://github.com/elastic/kibana/issues/124805

## To Verify
* Create a rule that performs an ES search and let it run. Note that we cannot use the index threshold, ES query, or any security solutions rules because they are using the abortable search client which this PR does not touch. Alternatively, you can switch one of those rule types back to use the normal ES client for testing.
* Verify that `execute` events in the event log have new fields added under `kibana.alert.rule.execution.metrics` to keep track of these metrics.

### Checklist
